### ISSUE TITLE
Re-evaluate the use of openfoodfacts-build-cache

### ISSUE THREAD

**Issue by alexgarel** [openfoodfacts/openfoodfacts-server#9220]
We are using a git repository as a cache for our taxonomy builds.

So far so good. But it is growing at a quite big pace (more than 2000 files in the taxonomies directory).

I'm not sure how long we should go with it but I propose to be able to clean it up from times to times.

I think this would mean:
* having a CI task to:
  * (easy) remove old files from time to time,
  * (harder) and eventually cleanup history and git force push a new main branch (but instances pulling from them must ensure they pull with option to avoid being stuck by this)
* to avoid loosing taxonomies linked to releases, we should attach corresponding compiled taxonomies in release, in a tar file (you can see how [it is done for dist files](https://github.com/openfoodfacts/openfoodfacts-server/blob/main/.github/workflows/release-please.yml#L77))

**Comment by alexgarel** [comment]
@john-gom and @stephanegigandet, what do you think ? (It's not that urgent)

**Comment by john-gom** [comment]
Regarding the concern around losing taxonomies linked to releases, could we not just look in the image history? https://github.com/openfoodfacts/openfoodfacts-server/pkgs/container/openfoodfacts-server%2Fbackend/versions?filters%5Bversion_type%5D=tagged

Regarding cleaning up, I would be tempted to go with the easy option for now. I feel like the hard option might be superseded by other initiatives, e.g. if the taxonomy editor or some similar project gets to completion then it might make more sense for that project to build the taxonomies as an "image" so that PO can just download them.

I'd also still like to persuade @stephanegigandet that we shouldn't be using real taxonomies in our tests, but test taxonomies with specifically designed test data to exercise the desired parts of the code. If we want to specifically run tests on the real taxonomies then maybe that should be part of the taxonomy editor project (or ideally not allow users to make errors in the first place).

**Comment by stephanegigandet** [comment]
@alexgarel I don't think we really need to keep built taxonomies for old releases. They can always be rebuilt if for some reason they are needed.
